static/app/utils/metrics/metricsProvider.tsx

```diff
@@ -0,0 +1,121 @@
+import {useEffect, useState} from 'react';
+
+import {addErrorMessage} from 'sentry/actionCreators/indicator';
+import {Client} from 'sentry/api';
+import {t} from 'sentry/locale';
+import {MetricsMetaCollection, MetricsTagCollection, Organization} from 'sentry/types';
+import handleXhrErrorResponse from 'sentry/utils/handleXhrErrorResponse';
+import useApi from 'sentry/utils/useApi';
+
+import {MetricsContext, MetricsContextValue} from './metricsContext';
+
+async function fetchMetricMetas(
+  api: Client,
+  organization: Organization,
+  projects: number[]
+): Promise<MetricsMetaCollection> {
+  return api.requestPromise(`/organizations/${organization.slug}/metrics/meta/`, {
+    query: {
+      project: projects,
+    },
+  });
+}
+
+async function fetchMetricTags(
+  api: Client,
+  organization: Organization,
+  projects: number[],
+  fields: string[]
+): Promise<MetricsTagCollection> {
+  return api.requestPromise(`/organizations/${organization.slug}/metrics/tags/`, {
+    query: {
+      project: projects,
+      metrics: fields,
+    },
+  });
+}
+
+type MetricsProviderProps = {
+  children: React.ReactNode | ((props: MetricsContextValue) => React.ReactNode);
+  fields: string[];
+  organization: Organization;
+  projects: number[];
+  skipLoad?: boolean;
+};
+
+export function MetricsProvider({
+  children,
+  projects,
+  fields,
+  organization,
+  skipLoad = false,
+}: MetricsProviderProps) {
+  const api = useApi();
+  const [state, setState] = useState({metas: {}, tags: {}});
+
+  useEffect(() => {
+    if (!skipLoad) {
```

Comment:
Should it be the other way around? return undefined if skip?

Reply:
yes, we should not have the `!`... I believe Jonas missed this in his commit